app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsViewModel.kt

```diff
@@ -133,60 +69,140 @@ class SessionDetailsViewModel @JvmOverloads constructor(
     val subtitleFont = Font.Roboto.Light
     val titleFont = Font.Roboto.BoldCondensed
 
-    fun onOptionsMenuItemSelected(menuItemId: Int) = when (menuItemId) {
-        R.id.menu_item_feedback -> {
-            val uri = feedbackUrl.toUri()
-            viewActionHandler.openFeedback(uri)
-            true
-        }
-        R.id.menu_item_share_session,
-        R.id.menu_item_share_session_text -> {
-            val formattedSession = session.toPlainText(timeZoneId)
-            viewActionHandler.shareAsPlainText(formattedSession)
-            true
+    val selectedSessionParameter: LiveData<SelectedSessionParameter> = repository.selectedSession
+        .map { it.toSelectedSessionParameter() }
+        .asLiveData(executionContext.database)
+
+    val openFeedBack = SingleLiveEvent<Uri>()
+    val shareSimple = SingleLiveEvent<String>()
+    val shareJson = SingleLiveEvent<String>()
+    val addToCalendar = SingleLiveEvent<Session>()
+    val setAlarm = SingleLiveEvent<Unit>()
+    val navigateToRoom = SingleLiveEvent<Uri>()
+    val closeDetails = SingleLiveEvent<Unit>()
+
+    private fun Session.toSelectedSessionParameter(): SelectedSessionParameter {
+        val useDeviceTimeZone = repository.readUseDeviceTimeZoneEnabled()
+        val formattedZonedDateTime = formattingDelegate.getFormattedDateTimeShort(useDeviceTimeZone, dateUTC, timeZoneOffset)
+        val formattedAbstract = markdownConversion.markdownLinksToHtmlLinks(abstractt)
+        val formattedDescription = markdownConversion.markdownLinksToHtmlLinks(description)
+        // language=regex
```

Comment:
Is there a need to leave that?

Reply:
Yes. It is used in Android Studio to correctly syntax highlight the regular expression part of the statement. Now moved to `SessionFormatter`.